src/net_processing.h

```diff
@@ -143,6 +143,10 @@ class PeerManager final : public CValidationInterface, public NetEventsInterface
      *  May return nullptr if the Peer object can't be found. */
     PeerRef GetPeerRef(NodeId id) const;
 
+    /** Get a shared pointer to the Peer object and remove it from m_peer_map.
+     *  May return nullptr if the Peer object can't be found. */
```

Comment:
aedd418fed7abfd00cc731bd57ed5b013259eb5a

To be precise, it may return an [empty](https://en.cppreference.com/w/cpp/memory/shared_ptr/shared_ptr) shared pointer rather `nullptr`.

Reply:
fixed